lib/manageiq/util/memory_logging.rb

```diff
@@ -0,0 +1,41 @@
+require 'miq-process'
+
+module ManageIQ::Util::MemoryLogging
+  extend ActiveSupport::Concern
+  include Vmdb::Logging
+
+  def memory_logger(message, &block)
+    debug(message) if block
+    yield
+    debug(message)
+  end
+
+  def memory_logger_with_gc(message, &block)
+    debug(message) if block
+    yield
+    GC.start
+    GC.start
+    GC.start
```

Comment:
I still have this thing from years ago that supposedly makes GC.start more likely to do it's thing.  I don't know if we want it.   cc @Fryguy 

```ruby
def iter n
  if n > 0
    1.times{
      iter(n-1)
    }
  else
    # null code
  end
end

def gc
  iter(100)
  GC.start
  GC.start
end
```

Reply:
Does the `1.times` have any use other than making the stacklevel larger?

Reply:
@NickLaMuro If I knew, I would attempt to explain it.  I don't remember if Aaron or Matthew sent it to me but they passed it along from someone else (maybe Koichi) and that it supposedly worked better to ensure GC worked.  It's like a fable passed down through the generations of developers.

Reply:
👵 👴 :book 👨 👩

Reply:
It's funny though, I think we need to `GC.start` 3 times to make young objects old... I'm not sure if Aesop's GC.start above handles young->old in the same way.

Reply:
[3 times GC.start link](https://github.com/ko1/nakayoshi_fork#mechanism)

I'm fine with keeping what we have here.  I just remembered that code from Aaron/Matthew and thought it was relevant.